pages/preview.tsx

```diff
@@ -0,0 +1,118 @@
+/**
+ * @file e.tsx
+ * Exports the Embed page component.
+ */
+
+import type { GetServerSideProps } from 'next';
+import type { IRss } from '@interfaces/data';
+import type { IPageError } from '@interfaces/error';
+import type { IPreviewPageProps } from '@interfaces/page';
+import { useEffect, useState } from 'react';
+import Head from 'next/head';
+import Error from 'next/error';
+import parseEmbedParamsToConfig from '@lib/parse/config/parseEmbedParamsToConfig';
+import fetchRssFeed from '@lib/fetch/rss/fetchRssFeed';
+import parseEmbedData from '@lib/parse/data/parseEmbedData';
+import Embed from '@components/Embed/Embed';
+import generateEmbedUrl from '@lib/generate/string/generateEmbedUrl';
+import generateEmbedHtml, {
+  generateEmbedStyles,
+  getEmbedHeight
+} from '@lib/generate/html/generateEmbedHtml';
+
+const PreviewPage = ({ config, rssData, error }: IPreviewPageProps) => {
+  const [newConfig, setNewConfig] = useState(config);
+  const data = parseEmbedData(newConfig, rssData);
+  const embedData = {
+    ...data,
+    ...(newConfig.showPlaylist &&
+      newConfig.showPlaylist !== 'all' &&
+      newConfig.showPlaylist > 1 &&
+      data.playlist && {
+        playlist: data.playlist.slice(0, newConfig.showPlaylist)
+      })
+  };
+
+  function handlePostMessage(e: MessageEvent) {
+    if (
+      !/^https?:\/\/(feeder)(\.staging)?\.prx\.(org|tech|test)$/.test(e.origin)
+    )
+      return;
+
+    setNewConfig((prevConfig) => ({
+      ...prevConfig,
+      ...(e.data || {})
+    }));
+  }
+
+  useEffect(() => {
+    const postMessageData = {
+      embedUrl: generateEmbedUrl(newConfig),
+      embedHtml: generateEmbedHtml(newConfig),
+      embedHeight: getEmbedHeight(newConfig),
+      embedStyles: generateEmbedStyles(newConfig)
+    };
+
+    window.parent.postMessage(postMessageData, '*');
+  }, [newConfig]);
+
+  useEffect(() => {
+    window.addEventListener('message', handlePostMessage);
+
+    return () => {
+      window.removeEventListener('message', handlePostMessage);
+    };
+  }, []);
+
+  if (error) {
+    return <Error statusCode={error.statusCode} title={error.message} />;
+  }
+
+  return (
+    <>
+      <Head>
+        <title>PRX Play - Embeddable Player</title>
+      </Head>
+      <Embed config={newConfig} data={embedData} />
+    </>
+  );
+};
+
+export const getServerSideProps: GetServerSideProps<
+  IPreviewPageProps
+> = async ({ query, res }) => {
+  // 1. Convert query params into embed config.
```

Comment:
So how much of this code is duplicative of `pages/embed.tsx`?  Just seems like a bit more lines of code than I would expect, for a page that's nearly identical.